src/cursor/abstract_cursor.ts

```diff
@@ -457,9 +458,24 @@ export abstract class AbstractCursor<
    * cursor.rewind() can be used to reset the cursor.
    */
   async toArray(): Promise<TSchema[]> {
-    const array = [];
+    const array: TSchema[] = [];
+
+    // when each loop iteration ends,documents will be empty and a 'await (const document of this)' will run a getMore operation
     for await (const document of this) {
       array.push(document);
+      let docs = this.readBufferedDocuments();
+      if (this.transform != null) {
+        docs = await Promise.all(
+          docs.map(async doc => {
+            if (doc != null) {
+              return await this.transformDocument(doc);
```

Comment:
transformDocument is only async because it cleans up the cursor.  Does performance improve if we inline the call to `transform` here instead?

Reply:
That would prevent unnecessary promises from being constructed.

Reply:
Our benchmarks do not run a cursor with a transform set but it would be great to get an idea of "how bad" this way of writing the code is.

Reply:
I ran tests with a transform and without, as well as making `transformDocuments` sync/async, doesn't seem like there's any local performance changes either way. Here are my [results](https://docs.google.com/document/d/1hm0cKfyz3FTYqQIIwY1pnQsBdunJ0ZGrmWF1UbAcnOI/edit#heading=h.8yw5ksibnwoj).